### PR TITLE
http/file_server.ts  fix error conditional response 

### PR DESCRIPTION
**Pull request by masx200**
fix error conditional response 

https://github.com/denoland/deno_std/issues/2207

### CODE REVIEW

**Review by kt3k — COMMENTED** [review]
@masx200 
Thanks for the fix! Can you also add a test case similar to the case `file_server returns 304 for requests with if-none-match set with the etag` but with `W/` prefixed etag in request headers.

**Comment by kt3k** [comment]
and also can you run `deno fmt` to the changed source code?

File: http/file_server_test.ts [hunk 1]
```diff
@@ -991,3 +991,94 @@ Deno.test(
...
+      assert(etag.length > 0);
+      assert(etag.startsWith("W/"));
+      {
+        const res = await fetch(
+          testurl,
+          {
+            headers: {
+              "if-none-match": etag,
+
+              "If-Modified-Since": last_modified,
```
Comment (kt3k) [inline http/file_server_test.ts:1067]: I think there's no need to specify `If-Modified-Since` when etag is already set

File: http/file_server_test.ts [hunk 2]
```diff
@@ -991,3 +991,94 @@ Deno.test(
...
+    try {
+      const testurl = "http://localhost:4507/testdata/desktop.ini";
+      const fileurl = new URL("./testdata/desktop.ini", import.meta.url);
+      let etag: string | undefined | null;
+      let last_modified: string | null | undefined;
+      {
+        const res = await fetch(
+          testurl,
+          {
+            headers: [
```
Comment (kt3k) [inline http/file_server_test.ts:1008]: Most headers look irrelevant to the purpose of the test case. I guess we need only `Accept-Encoding` to get `W/` prefixed etag.

**Review by kt3k — APPROVED** [review]
Thanks for updating the test case!

LGTM
